server/src/test/java/io/crate/integrationtests/ViewsITest.java

```diff
@@ -303,4 +303,21 @@ public void test_object_column_of_view_contain_inner_types() throws Exception {
             "o['value_2']| integer"
         );
     }
+
+    /*
+     * https://github.com/crate/crate/issues/17836
+     */
+    @Test
+    public void test_error_on_unknown_object_key_on_authenticated_user() {
+        try (var session = sqlExecutor.newSession()) {
+            execute("SET error_on_unknown_object_key=false", session);
+            execute("CREATE TABLE doc.tbl1 (obj OBJECT(DYNAMIC))", session);
+            execute("CREATE VIEW vw1 AS SELECT obj['not_existing'] FROM doc.tbl1;", session);
+            execute("select view_definition from information_schema.views where table_name='vw1';", session);
+            assertThat(response).hasRows(
+                "SELECT \"obj\"['not_existing']\n" +
+                    "FROM \"doc\".\"tbl1\"\n"
+            );
```

Comment:
Don't we need to reset the setting and execute the view to verify the fix?

Reply:
If I understood it correctly, the approach is to save the setting that was used at view creation, and always use this, and not the session setting of the user executing the view, or I get it wrong, we want both?

Reply:
Right, but the current test is not verifying this.

Reply:
After fixing the test, it highlights that it's not yet working as expected. I'll fix it.